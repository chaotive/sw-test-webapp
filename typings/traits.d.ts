export interface IPaginable {
  goBack: () => void
  goForward: () => void
  state: {
    previousPage?: string
    nextPage?: string
  }
}
