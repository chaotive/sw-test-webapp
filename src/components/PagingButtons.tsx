import React from 'react'
import {IPaginable} from '../../typings/traits'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Button from 'react-bootstrap/Button'

const PagingButtons: React.FC<IPaginable> = props => <Row>
  <Col><Button disabled={!props.state.previousPage} onClick={props.goBack}>Back</Button></Col>
  <Col><Button disabled={!props.state.nextPage} className="float-right" onClick={props.goForward}>Forward</Button></Col>
</Row>

export default PagingButtons
