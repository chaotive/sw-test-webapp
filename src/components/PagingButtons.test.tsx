import React from 'react'
import ReactDOM from 'react-dom'
import {IPaginable} from '../../typings/traits'
import PagingButtons from './PagingButtons'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const testData: IPaginable = {
    goBack: () => undefined,
    goForward: () => undefined,
    state: {
      previousPage: 'someUrl',
      nextPage: 'anotherUrl'
    }
  }

  const div = document.createElement('div')
  ReactDOM.render(<PagingButtons {...testData} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly for empty state property in props', () => {
  const testData: IPaginable = {
    goBack: () => undefined,
    goForward: () => undefined,
    state: {}
  }

  const tree = renderer
    .create(<PagingButtons {...testData} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly for existing state property in props', () => {
  const testData: IPaginable = {
    goBack: () => undefined,
    goForward: () => undefined,
    state: {
      previousPage: 'someUrl',
      nextPage: 'anotherUrl'
    }
  }

  const tree = renderer
    .create(<PagingButtons {...testData} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
