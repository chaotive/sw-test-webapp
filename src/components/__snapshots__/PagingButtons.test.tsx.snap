// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for empty state property in props 1`] = `
<div
  className="m-4 row"
>
  <div
    className="col"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </div>
  <div
    className="col"
  >
    <button
      className="float-right btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Forward
    </button>
  </div>
</div>
`;

exports[`renders correctly for existing state property in props 1`] = `
<div
  className="m-4 row"
>
  <div
    className="col"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </div>
  <div
    className="col"
  >
    <button
      className="float-right btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Forward
    </button>
  </div>
</div>
`;
